src/fpm_targets.f90

```diff
@@ -238,6 +240,17 @@ subroutine build_target_list(targets,model)
                         call add_dependency(targets(1)%ptr, targets(size(targets))%ptr)
                     end if
 
+                case (FPM_UNIT_CPPSOURCE) 
+
+                    call add_target(targets,package=model%packages(j)%name,source = sources(i), &
+                                type = FPM_UNIT_CPPSOURCE, &
```

Comment:
```suggestion
                                type = FPM_TARGET_CPP_OBJECT, &
```

Reply:
When I made this line change, fpm started to break when compiling .cpp files.

`g++ -c ./src/hello_world.cpp  -o build/gfortran_00000000811C9DC5/cpp_files/src_hello_world.cpp.o` 

This above command is a valid command. Right ?

cpp_files.f90                          done.
hello_world.cpp                        compiling...
[ 33%] Compiling... 
Fortran runtime error: EXECUTE_COMMAND_LINE: Invalid command line

Error termination. Backtrace:
#0  0x7fc4c1caeae0 in ???
#1  0x7fc4c1caf659 in ???
#2  0x7fc4c1cafb09 in ???
#3  0x7fc4c1f2183c in ???
#4  0x7fc4c1f21ac9 in ???
#5  0x7fc4c1f21b8f in ???
#6  0x555bea2e3b15 in __fpm_filesystem_MOD_run
        at ./src/fpm_filesystem.F90:907
#7  0x555bea35e830 in __fpm_compiler_MOD_compile_cpp
        at ./src/fpm_compiler.f90:934
#8  0x555bea368c3f in build_target
        at ./src/fpm_backend.F90:329
#9  0x555bea36c745 in __fpm_backend_MOD_build_package
        at ./src/fpm_backend.F90:126
#10  0x555bea2c01b1 in __fpm_MOD_cmd_build
        at ./src/fpm.f90:310
#11  0x555bea2b516e in MAIN__
        at app/main.f90:68
#12  0x555bea2b56e8 in main
        at app/main.f90:11

Reply:
Yes, it should work. What happens when you try to run that command directly in your terminal?

Reply:
My bad, I forgot that I reinstalled Ubuntu on my PC which means g++ is not installed on my system. I had to install it, and it is now working fine.